library/src/meta/reference.rs

```diff
@@ -173,3 +193,30 @@ cast_to_value! {
         Supplement::Func(v) => v.into(),
     }
 }
+
+/// A citable element can impl this trait to set the supplement content
+/// when it be referenced.
+pub trait ReferenceInfo {
+    /// The counter used in reference.
+    fn counter(&self, styles: StyleChain) -> Option<Counter>;
+
+    /// The numbering used in reference.
+    fn numbering(&self, styles: StyleChain) -> Option<Numbering>;
+
+    /// supplement used in reference.
+    fn supplement(&self, _styles: StyleChain) -> Option<Supplement>;
+
+    // default logic of convert supplement into content in reference
+    fn resolve_supplement(
```

Comment:
this is only used once right? if so, no need to put it into the trait.

Reply:
Thinking of that, there are more places where in might be useful (outline mostly), additionally some objects in the future might want to override their supplement logic.

Reply:
As I will rework `RefInfo` to make it more useful and try and nicely include functionality of `RefElem` in a nicer package, this function will get reworked either way